Update the blocks to the latest version of Gutenberg

kienstra opened the pull request:
* Updates this theme to the changes in Gutenberg
* Before this, most of the blocks had errors due to Gutenberg changes.

kienstra committed:
Change the Activty List block to use <ServerSideRender>

This will be more DRY,
and it won't require refactoring
to accomodate the removal of withAPIData from Gutenberg.

kienstra committed:
Update Activity List to Gutenberg changes

Also use the <ServerSideRender> component.
This is a dynamic block.

kienstra committed:
Update Discover block, using ServerSideRender

This is a dynamic block,
so this will enable showing the changes dynamically.

kienstra committed:
Update Featured Destinations block to new API

There's still work needed in the styling,
as the block seems to overflow the container.

kienstra committed:
Simple update to the Hero block

Simply change the source of
the RichText component.

kienstra committed:
Update Adventures block, using ServerSideRender

@todo: some kind of placeholder in case
there aren't enough adventures.

kienstra committed:
Make very simple changes to the Search block

Use wp.editor to get RichText.
Also, change the title to "Travel Search".
There's another search block.

kienstra committed:
Make the const declarations alphabetical

Also, remove an empty file:
travel-angles/index.js

kienstra committed:
Styling adjustments, especially for Top Adventures and Featured Destinations

Make styling changes in the editor,
and also output <img> instead of <amp-img>
The plugin will convert these to <amp-img>.

kienstra committed:
Re-add the placeholder for the Popular Adventures block

There was a JS-based placeholder before.
But now, output this in the PHP render_callback.

kienstra committed:
Address Travis failure by changing the install source

This might not work,
but change the source to what appears to be
the new path to travis.install.sh

kienstra committed:
Attempt to address Travis failure againa

Add /scripts to DEV_LIB_PATH
It seems that there's a script in wp-dev-lib
that expects DEB_LIB_PATH to be the /scripts directory.

kienstra committed:
Update composer.json for Travis failure

This should use phpcompatibility instead of
wimg/php-compatibility

kienstra committed:
Set the DEV_LIB_PATH in the env section

In an attempt to fix the
failed Travis builds.

kienstra committed:
Move DEV_LIB_PATH back under install

kienstra commented on includes/class-amp-travel-blocks.php:
```diff
@@ -508,6 +519,9 @@ public function render_block_travel_featured( $attributes ) {
 		if ( count( $locations ) !== self::FEATURED_LOCATIONS_COUNT ) {
+			if ( $is_server_side_render ) {
+				return __( 'Not enough featured locations found. Please add at least six "Locations" terms, select an image, and check "Featured destination."', 'travel' );
+			}
```
This implements the placeholder that used to be in the JavaScript `edit()` function. It's not really ideal.

It'd be nice if there was a way to detect if the `<ServerSideRender>` component didn't render anything, so this placeholder text didn't have to be in this PHP `render_callback`. 

Maybe there's a way to do that.

miina replied:
It should be sufficient for the purposes of the demo.

kienstra replied:
Nice, thanks

kienstra committed:
Attempt to address the failed Travis build

commit composer.lock,
as phpcompatibility was not recognized.

kienstra committed:
Move wp-dev-lib from package.json to composer.json

kienstra committed:
Update composer.lock with composer update

kienstra commented on blocks/search/index.js:
```diff
@@ -7,16 +7,17 @@
 	'amp-travel/search',
 	{
-		title: __( 'Search' ),
+		title: __( 'Travel Search' ),
```
There's a 'Search' block in Core, so maybe this will help to distinguish it.

miina replied:
Ideally we could also create a separate category for Travel blocks, but I think it's not that critical for the demo purposes at this moment.

kienstra committed:
Remove composer.lock from this PR

It might not be a bad idea,
but as long as the build passes without it,
it's probably outside of the scope.

kienstra committed:
Address a PHPCS error of not enough tabs.

kienstra commented on .travis.yml:
```diff
@@ -36,8 +36,8 @@ env:
 install:
-  - npm install
-  - export DEV_LIB_PATH=node_modules/wp-dev-lib
+  - composer install
```
The Travis builds were failing, so I eventually moved `wp-dev-lib` to `composer`.

kienstra commented on blocks/activity-list/index.js:
```diff
@@ -1,83 +1,55 @@
+			__( 'Travel', 'travel' )
 		],
 
-		edit: withAPIData( () => {
```
`withAPIData` was deprecated (removed). There is `wp.apiFetch()`, but that returns a promise. I thought it'd be simpler to use the `ServerSideRender` component instead.

kienstra commented on composer.json:
```diff
@@ -4,8 +4,9 @@
+        "xwp/wp-dev-lib": "^1.0",
         "wp-coding-standards/wpcs": "^0.14.0",
-        "dealerdirect/phpcodesniffer-composer-installer": "^0.4.4",
-        "wimg/php-compatibility": "^8.1"
```
`wimg/php-compatibility` looks to be deprecated or discontinued.

kienstra committed:
Add space inside brackets in Popular Adventures block

kienstra commented:
**Request For Code Review**

Hi @miina,
Would you be available to review this?

For the dynamic blocks, this uses `ServerSideRender`. 

If you still have this repo set up, please re-run `composer install && npm install && npm run dev`

Here's the display in the block editor with this PR:

![travel-blocks](https://user-images.githubusercontent.com/4063887/55456803-3d6c8e00-55a5-11e9-9dae-715b14e28c0b.gif)

Thanks, Miina! No problem if you're really busy, I could see if someone else is available.

miina commented:
Sure, @kienstra, I'll review it shortly.

miina commented on blocks/activity-list/index.js:
```diff
@@ -1,83 +1,55 @@
-					</div>
-				</section>
+
+				<Fragment key="activity-ssr">
```
We could wrap this into `<Disabled>` instead to prevent interacting with the view and being directed away when clicking on links.

kienstra replied:
Good idea, https://github.com/xwp/travel/pull/86/commits/4ce2b7fe1257973b51832b298045176606075d77 adds `<Disabled>`

miina commented on blocks/featured/index.js:
```diff
@@ -1,146 +1,56 @@
-						</div>
-					</div>
-				</section>
+				<Fragment key="discover-ssr">
```
Same here about `<Disabled>`.

kienstra replied:
Sure, https://github.com/xwp/travel/pull/86/commits/4ce2b7fe1257973b51832b298045176606075d77 adds `<Disabled>`

miina reviewed (commented):
Thank you for reworking all this. Looks good for the demo.

The only item that maybe could be changed for the demo as well could be trying using `Disabled` component around `ServerSideRender` to avoid clicking on links. Otherwise, I think it's good to go 👍

kienstra committed:
Wrap Activity List and Featured blocks in Disabled

Wrap the Fragment outside of the ServerSideRender,
though it still looks like it's possible to click.

kienstra commented:
**Thanks For Reviewing**

Hi @miina,
Thanks for reviewing this.

https://github.com/xwp/travel/pull/86/commits/4ce2b7fe1257973b51832b298045176606075d77 adds `<Disabled>`, though at least in my local in looks like it's still possible to click the links and go to another page, like an 'Activity' link in the 'Activity List' block.

miina commented:
@kienstra Okay, good to know, I had the impression that it would remove pointer events but looks like it might just disable buttons then 🤷‍♀️ 

Should be ready for merging then.

kienstra commented:
Thanks, it might just be my local 😄

kienstra commented:
Thanks, @miina! Could you please approve this if you're ready?

miina reviewed (approved):
LGTM!

kienstra commented:
Thanks!
